### ISSUE TITLE
Doc argument in Error Handling Middleware is null in 4.11

### ISSUE THREAD

**Issue by JustinLivi** [Automattic/mongoose#5496]
**Do you want to request a *feature* or report a *bug*?**

Report a bug

**What is the current behavior?**

In Error Handling Middleware `doc` argument is `null` when using 4.11.3

```javascript
schema.post('save', function (error, doc, next) {
    console.log(error); // E11000 duplicate key error
    console.log(doc); // null <---
    next(error);
});
```

**If the current behavior is a bug, please provide the steps to reproduce.**

- consistently seeing this issue when triggering duplicate key errors in mongo
- have not tested other types of errors
- downgrading to 4.10.8 resolves the issue

**What is the expected behavior?**

The `doc` argument should not be `null` . It should be the document which caused the error.

Behavior in 4.10.8:
```javascript
schema.post('save', function (error, doc, next) {
    console.log(error); // E11000 duplicate key error
    console.log(doc); // the actual doc <---
    next(error);
});
```

**Please mention your node.js, mongoose and MongoDB version.**

node v8.1.4
mongoose v4.11.3
mongodb v3.4.5

**Comment by davebaol** [comment]
Already fixed in 4.11.4-pre, see issue https://github.com/Automattic/mongoose/issues/5466

**Comment by varunjayaraman** [comment]
^yep, closing this as a duplicate
